luwak/src/main/java/uk/co/flax/luwak/Monitor.java

```diff
@@ -693,14 +691,12 @@ public boolean needsIntervals() {
 
         @Override
         public void collectLeafPosition(Scorer scorer, Interval interval, int docID) {
-            String terms = document.getField(interval.field).stringValue();
-            if (!matchingTerms.containsKey(currentId))
-                matchingTerms.put(currentId, new StringBuilder());
-            matchingTerms.get(currentId)
-                    .append(" ")
-                    .append(interval.field)
-                    .append(":")
-                    .append(terms.substring(interval.offsetBegin, interval.offsetEnd));
+            Query query = scorer.getWeight().getQuery();
+            if (query instanceof TermQuery) {
+                if (!matchingTerms.containsKey(currentId))
+                    matchingTerms.put(currentId, new StringBuilder());
+                matchingTerms.get(currentId).append(" ").append(((TermQuery) query).getTerm());
```

Comment:
Right.  Although it might be nice to make it more useful, maybe returning a Set<Term> instead of the stringbuilder.